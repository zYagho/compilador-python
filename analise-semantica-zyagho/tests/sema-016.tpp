{Erro: Função 'func' deveria retornar 'vazio', mas retorna 'flutuante'}
{Aviso: Variável 'b' declarada e inicializada, mas não utilizada}
{Erro: Função 'principal' deveria retornar 'inteiro', mas retorna 'vazio'}

flutuante: a
inteiro: b

func()
  a := 10.2
  retorna(a)
fim

inteiro principal()
	b := 5
	func()
fim
