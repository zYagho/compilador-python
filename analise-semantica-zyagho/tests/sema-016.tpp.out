ERR-SEM-FUNC-RET-TYPE-ERROR
WAR-SEM-VAR-DECL-INIT-NOT-USED
ERR-SEM-FUNC-RET-TYPE-ERROR
